crates/explorer/src/index.js

```diff
@@ -189,8 +93,98 @@ const renderInst = (mnemonic, operands) => {
   }
 };
 
-// Render the ASM.
+const linkElements = (element) => {
+  const selector = (offset) =>
+    document.querySelectorAll(`[data-wasm-offset="${offset}"]`);
+  const eachElementWithSameWasmOff = (event, closure) => {
+    let offset = event.target.dataset.wasmOffset;
+    if (offset !== null) {
+      for (const elem of selector(offset)) closure(elem);
```

Comment:
Walking the DOM every time seems more expensive than the old approach of maintaining a map from offset to list of elements associated with that offset. I'd prefer sticking with the previous approach unless there is strong motivation to do otherwise.